gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -288,6 +303,15 @@ private static void modifyStateIfRetryRequired(org.apache.gobblin.configuration.
     state.removeProp(TimingEvent.FlowEventConstants.DOES_CANCELED_FLOW_MERIT_RETRY);
   }
 
+  private static boolean isStateTransitionToFinal(org.apache.gobblin.configuration.State currentState, List<org.apache.gobblin.configuration.State> prevStates) {
+    Set<String> finalStates = ImmutableSet.of(ExecutionStatus.COMPLETE.name(), ExecutionStatus.CANCELLED.name(), ExecutionStatus.FAILED.name());
```

Comment:
may be better as a class `static`

Reply:
still thinking the same... why initialize on every invocation, rather than once when the class is loaded?

Reply:
I think we have FlowStatusGenerator.FINISHED_STATUSES which is essentially the same list, maybe reuse that one?